ai_summary/indico_ai_summary/controllers.py

```diff
@@ -0,0 +1,110 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2025 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from flask import Response, jsonify, stream_with_context
+from flask_pluginengine import current_plugin
+from webargs import fields
+from webargs.flaskparser import use_kwargs
+
+from indico.core.db import db
+from indico.core.errors import IndicoError
+from indico.modules.categories.controllers.base import RHManageCategoryBase
+from indico.modules.events.management.controllers.base import RHManageEventBase
+from indico.modules.events.notes.util import get_scheduled_notes
+
+from indico_ai_summary.llm_interface import LLMInterface
+from indico_ai_summary.models.prompt import Prompt
+from indico_ai_summary.schemas import PromptSchema
+from indico_ai_summary.utils import MarkupMode, chunk_text, convert_markup, generate_chunk_stream
+from indico_ai_summary.views import WPCategoryManagePrompts
+
+
+CATEGORY_SIDEMENU_ITEM = 'plugin_ai_summary_prompts'
+
+
+class RHManageCategoryPrompts(RHManageCategoryBase):
+
+    def _process_args(self):
+        RHManageCategoryBase._process_args(self)
+
+    def _process_GET(self):
+        prompts = Prompt.query.with_parent(self.category).all()
+        serialized_prompts = PromptSchema(many=True).dump(prompts)
+        return WPCategoryManagePrompts.render_template('manage_category_prompts.html', category=self.category,
+                                                       active_menu_item=CATEGORY_SIDEMENU_ITEM,
+                                                       stored_prompts=serialized_prompts)
+
+    @use_kwargs({'prompts': fields.List(fields.Nested(PromptSchema), required=True)}, location='json')
+    def _process_POST(self, prompts):
+        Prompt.query.with_parent(self.category).delete()
+        for prompt_data in prompts:
+            prompt = Prompt(category_id=self.category.id, name=prompt_data['name'], text=prompt_data['text'])
+            db.session.add(prompt)
+        return '', 204
+
+
+class SummarizeEvent(RHManageEventBase):
+
+    CSRF_ENABLED = False
+
+    def _process_args(self):
+        RHManageEventBase._process_args(self)
+
+    @use_kwargs({'prompt': fields.Str(load_default='')}, location='json')
```

Comment:
this should be `required=True` and not have a default (and json/post is the default location, so you can remove it)